### PR TITLE
SECURITY-176 User-supplied Groovy code run in form validation

### PR DESCRIPTION
**Pull request by anuragsen**
Change-Id: Id7f2acfe723f7e389997cc1b8eec0c96e183bc3e

### CODE REVIEW

File: src/main/java/hudson/scm/IntegrityCheckpointAction.java [hunk 1]
```diff
@@ -74,16 +79,17 @@ public IntegrityCheckpointAction(String serverConfig, String checkpointLabel)
...
-    binding.setVariable("env", env);
-    binding.setVariable("sys", System.getProperties());
-    CompilerConfiguration config = new CompilerConfiguration();
-    // config.setDebug(true);
-    GroovyShell shell = new GroovyShell(binding, config);
-    Object result = shell.evaluate("return \"" + expression + "\"");
+	binding.setVariable("env", env);
+	// binding.setVariable("sys", System.getProperties());
+	ClassLoader loader = GroovySandbox
+			.createSecureClassLoader(Jenkins.getActiveInstance().getPluginManager().uberClassLoader);
```
Comment (jglick) [inline src/main/java/hudson/scm/IntegrityCheckpointAction.java]: `evaluate` already wraps the supplied loader, so you should not call `createSecureClassLoader` yourself in this case.

File: src/main/java/hudson/scm/IntegrityCheckpointAction.java [hunk 2]
```diff
@@ -92,6 +98,19 @@ public static String evalGroovyExpression(Map<String, String> env, String expres
...
+  /**
+	 * @author asen
+	 * Default Whitelist
+	 *
+	 */
+	@Extension
+	public static class DefaultWhitelist extends ProxyWhitelist {
+		public DefaultWhitelist() throws IOException {
+			super(new StaticWhitelist("method java.text.Format format java.lang.Object",
+					"new java.text.SimpleDateFormat java.lang.String"));
```
Comment (jglick) [inline src/main/java/hudson/scm/IntegrityCheckpointAction.java]: Add as a PR to `script-security-plugin` if not already there.

File: src/main/java/hudson/scm/IntegrityCheckinAction.java [hunk 3]
```diff
@@ -207,8 +207,15 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
...
 
     // Create our Integrity check-in task
-    IntegrityCheckinTask ciTask = new IntegrityCheckinTask(ciConfigPath, ciWorkspaceDir, includes,
-        excludes, build, listener, getProjectSettings(build));
+    IntegrityCheckinTask ciTask;
+	try {
+		ciTask = new IntegrityCheckinTask(ciConfigPath, ciWorkspaceDir, includes,
+		    excludes, build, listener, getProjectSettings(build));
+	} catch (Exception e) {
+		LOGGER.fine("Exception Caught!  Skipping Integrity Checkin step! :"+ e.getMessage());
```
Comment (jglick) [inline src/main/java/hudson/scm/IntegrityCheckinAction.java]: More helpful to use `log(Level, String, Throwable)` to keep the stack trace.

File: src/main/java/hudson/scm/IntegrityCheckpointAction.java (hunk 2 above)
Comment (anuragsen) [reply src/main/java/hudson/scm/IntegrityCheckpointAction.java]: Dear @jglick ,
PR already submitted and closed for this. https://github.com/jenkinsci/script-security-plugin/pull/119/files. I will remove this in a subsequent plugin iteration.

File: src/main/java/hudson/scm/IntegrityCheckinAction.java (hunk 3 above)
Comment (anuragsen) [reply src/main/java/hudson/scm/IntegrityCheckinAction.java]: Dear @jglick,
Noted. Will be fixed in the next release

File: src/main/java/hudson/scm/IntegrityCheckpointAction.java (hunk 1 above)
Comment (anuragsen) [reply src/main/java/hudson/scm/IntegrityCheckpointAction.java]: Dear @jglick,
Noted. Will be fixed in the next release

**Pull request merged**